### PR TITLE
chore(deps): merge go-dep-parser into Trivy

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Merge go-dep-parser into Trivy

## Issues
- Close https://github.com/aquasecurity/trivy/issues/6105

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by nikpivkin — APPROVED** [review]
lgtm

File: pkg/dependency/parser/java/jar/sonatype/sonatype.go
Comment (DmitriyLewen) [inline pkg/dependency/parser/java/jar/sonatype/sonatype.go:1]: We migrated to trivy-java-db.
Do we need this now?

This job is not for this PR. Just as idea for next PR.

File: pkg/dependency/parser/types/types.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+type RefType string
+
+const (
+	RefWebsite      RefType = "website"
+	RefLicense      RefType = "license"
+	RefVCS          RefType = "vcs"
+	RefIssueTracker RefType = "issue-tracker"
+	RefOther        RefType = "other"
```
Comment (DmitriyLewen) [inline pkg/dependency/parser/types/types.go:76]: `RefWebsite`, `RefLicense` and `RefIssueTracker` are not used in Trivy.
Do you have info about using them outside of Trivy?

**Review by DmitriyLewen — COMMENTED** [review]
LGTM.
Left some small comments.

File: pkg/dependency/parser/java/jar/sonatype/sonatype.go
Comment (knqyf263) [reply pkg/dependency/parser/java/jar/sonatype/sonatype.go:1]: Yes, I also found it. We'll refactor it in another PR.

File: pkg/dependency/parser/types/types.go (hunk 1 above)
Comment (knqyf263) [reply pkg/dependency/parser/types/types.go]: I don't think they are used. Deleted https://github.com/aquasecurity/trivy/pull/6094/commits/4783137f9b99a3c6ae818e3e3559a1dd5095a7da

File: pkg/dependency/parser/io/io.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package io
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/dependency/parser/io/io.go:1]: We can move, but this PR is not refactoring anything other than linter errors. It just tries to merge go-dep-parser into Trivy.

File: pkg/log/logger.go
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/log/logger.go:1]: Fixed in https://github.com/aquasecurity/trivy/pull/6094/commits/906a1496ff59ae496a32c570c43aeec751da4263

[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/log/logger.go:1]: I'll remove a logger in go-dep-paser in another PR, though.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Comment by knqyf263** [comment]
I was supposed to keep git history, but the Trivy project allows only "squash and merge". I accidentally lost history of go-dep-parser😭

**Pull request merged**